src/openforms/payments/services.py

```diff
@@ -23,7 +23,7 @@ def update_submission_payment_registration(submission: Submission):
         return
 
     try:
-        plugin = register[submission.form.registration_backend]
+        plugin = register[submission.registration_backend.backend]
```

Comment:
I'm not sure if catching `AttributeError` here is the best option? It would silently fail typo's or other unintended `AttributeErrors`. I think I'd rather see something like:

```py
if (registration_backend := submission.registration_backend) is None:
    logevent.registration_payment_update_failure(submission, error=e)
    return
try:
    plugin = register[registration_backend.backend]
except KeyError as e:
    ...
```
the duplicated logevent line is annoying and I hope to introduce `structlog` which can de-duplicate this at some point.